sopel/modules/weather.py

```diff
@@ -16,16 +21,29 @@ def woeid_search(query):
     node for the result, so that location data can still be retrieved. Returns
     None if there is no result, or the woeid field is empty.
     """
-    query = 'q=select * from geo.placefinder where text="%s"' % query
+    query = 'q=select woeid from geo.places where text="%s"' % query
     body = web.get('http://query.yahooapis.com/v1/public/yql?' + query,
                    dont_decode=True)
     parsed = xmltodict.parse(body).get('query')
     results = parsed.get('results')
-    if results is None or results.get('Result') is None:
+    if type(results) is collections.OrderedDict:
+        place = results.get('place')
+        if type(place) is list:
+            return place[0]
+    if not (results):
+        return None
+    if type(results) is list:
+	first_result = result[0]
```

Comment:
There's a tab character here (and possibly also elsewhere?). Indentation must always be in multiples of four spaces. (It won't even load on Python 3 if you mix them.) Also, `result` isn't defined, and `first_result` is assigned to, but not used.